ui/src/components/form/text-field/text-input.tsx

```diff
@@ -0,0 +1,39 @@
+import { useComputedAtom } from '@principlestudios/jotai-react-signals';
+import { isAtom } from '@principlestudios/jotai-utilities/isAtom';
+import JotaiInput from '@principlestudios/react-jotai-form-components/input';
+import { useTwMerge } from '../../jotai/useTwMerge';
+import styles from './text-input.module.css';
+
+export function TextInput({
+	type = 'text',
+	className,
+	readOnly,
+	...props
+}: React.ComponentProps<typeof JotaiInput>) {
+	return (
+		<JotaiInput
+			className={useTwMerge(
+				styles.textInput,
+				'px-2 py-2 w-full',
+				'border',
+				useComputedAtom((get) =>
+					(isAtom(readOnly) ? get(readOnly) : readOnly)
```

Comment:
Nit: it may be worth making an exportable helper function for this behavior (`isAtom(x) ? get(x) : x`)?

i see it could be used in several places in this PR already

Reply:
Great idea, gonna actually add that to our library where `isAtom` comes from.

Reply:
See https://github.com/PrincipleStudios/jotai-react-signals/pull/18 - I will create a subsequent PR for this repo when that one is merged.